packages/netsuite-adapter/src/adapter.ts

```
@@ -266,7 +266,11 @@ export default class NetsuiteAdapter implements AdapterOperations {
       )
       : []
 
-    const dataElementsPromise = getDataElements(this.client, fetchQuery, this.getElemIdFunc)
+    const {
```

I've of course, tested this, and made sure this works as expected.
However, I would have expected `getConfigRecords` to fail earlier and not allow the code to pass all the way through here.

I also think we should probably change the following code to

https://github.com/salto-io/salto/blob/main/packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts#L348

```
async getConfigRecords() { 
       try { ...code } 
        catch (e) {
            log.error('getConfigRecords failed. received error: %s', e.message);
            // or e isInstanceOf InvalidSuiteAppCredentialsError
            if (e.message === 'Invalid SuiteApp credentials') {
                log.error('getConfigRecords failed with invalid creds error');
                throw e
            }
            return [];
        }

```


thoughts? 
I think probably doing both is legit, LMK your thoughts
(would love some directions here, as it's my first time touching the NS adapter code)

---

Nice catch @dorsev and thanks for taking it!
The reason we didn't have await here in first place is that we wanted to parallelize the calls to `getDataElements` (which perform "SuiteApp" calls) and `getCustomObjects` which performs SDF calls.

I wonder how we can make sure that we don't lose the advantage of parallelism while failing fast on invalid credentials.
The naive option would be to explicitly call `validateCredentials` at the beginning of the adapter's fetch, but it takes few seconds until it's done so if possible it's better to avoid that.
@dantal4 WDYT?

---

I see now that Daniel opened https://github.com/salto-io/salto/pull/5010/files, I will review it